projects/ADuCM360_demo_cn0409/src/CN0409.c

```diff
@@ -0,0 +1,745 @@
+*    contributors may be used to endorse or promote products derived
+*    from this software without specific prior written permission.
+*  - The use of this software may or may not infringe the patent rights
+*    of one or more patent holders.  This license does not release you
+*    from the requirement that you obtain separate licenses from these
+*    patent holders to use this software.
+*  - Use of the software either in source or binary form, must be run
+*    on or directly connected to an Analog Devices Inc. component.
+*
+* THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED
+* WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY
+* AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+* IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+* INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*
+*********************************************************************************/
+
+/***************************** Include Files **********************************/
+#include <stdio.h>
+#include <ADuCM360.h>
+#include <CN0409.h>
+#include "ADPD105.h"
+#include "Communication.h"
+#include "math.h"
+#include "Flash.h"
+#include "Timer.h"
+#include "stdlib.h"
+
+/********************************** Variables **********************************/
+float fpSlotACurrents[4] = {0,0,0,0};
+float fpSlotBCurrents[4] = {0,0,0,0};
+
+float fpFirstPoint, fpSecondPoint;
+
+float y_code[3], x_ntu[3], xm, ym;
+float num, denom;
+
+extern char s[100];
+
+CN0409_ISORatioCalibrationCoeff  ISORatioCoeff;
+CN0409_ISOCalibrationCoeff  ISONonRatioCoeff;
+CN0409_GLICoeff  GLICoeff;
+
+CN0409_NtuSolutions ntuSolutions;
+
+double firstRatio;
+double secondRatio;
+
+/*Variables for GLI2*/
+double firstSqrt;
+double secondSqrt;
+
+float fpTurbidityRatio;
+float fpTurbidityNonRatio;
+
+uint32_t  u32Buffer[5];
+
+/************************* Functions Definitions ******************************/
+
+/**
+   @brief Initialize the ADPD105 and prepare for measurement
+
+   @return void
+
+**/
+void CN0409_Init(void)
+{
+	/*Init CN0358*/
+	ADPD105_SetOperationMode(PROGRAM); //set programming mode
+
+	ADPD105_Reset();
+
+	ADPD105_SetOperationMode(PROGRAM); //set programming mode
+
+	ADPD105_SetCLK32K(1); //enable 32k clock
+	ADPD105_SetTimeSlotSwitch(PD_1_4_CNNECTED, PD_1_4_CNNECTED); //select photodiodes 1-4
+
+	/*Select LEDs for each time slot*/
+	ADPD105_SelectLED(LEDX2, SLOTA); //configure led1 for SlotA
+	ADPD105_SelectLED(LEDX1, SLOTB); //configure led1 for SlotB
+
+	/*Set led course value*/
```

Comment:
`/* Set led course value */ `-> add a blank space, please check this in the rest of the code as well